<div class="alert alert-info"> <b> 1. Знакомство с датасетом: </b>

In [1]:
import pandas as pd
import numpy as np
df = pd.read_csv('datasets_eat_places.csv')
df

Unnamed: 0                org_id city  average_bill    rating  \
0               1  16076540698036998306  msk         500.0  4.375000   
1               2   8129364761615040323  msk         500.0  4.000000   
2               4  13418544315327784420  msk         500.0  4.409091   
3               5   4997872545011433005  msk         500.0  4.541667   
4               8  11913692756639492916  spb         500.0  3.600000   
...           ...                   ...  ...           ...       ...   
43190       68333  17936653153885963296  msk           NaN  4.487952   
43191       68334   4379286080707082909  msk           NaN  3.812500   
43192       68335   7916477189329738565  msk           NaN  4.894231   
43193       68336  12358902585434046825  msk           NaN  4.156250   
43194       68338    125079240973229511  spb           NaN  4.633333   

                rubric  coffee_to_go  food_delivery  breakfast  \
0      Быстрое питание             1              1          0   
1              Кофейня             1              1          1   
2              Кофейня             1              1          1   
3              Кофейня             1              1          1   
4             Столовая             0              1          0   
...                ...           ...            ...        ...   
43190         Ресторан             1              1          1   
43191             Кафе             0              1          0   
43192         Ресторан             1              1          1   
43193             Кафе             1              1          1   
43194         Ресторан             0              0          0   

       summer_terrace  ...  business_lunch  payment_by_credit_card  car_park  \
0                   0  ...               0                       1         0   
1                   1  ...               1                       0         0   
2                   1  ...               1                       0         0   
3                   1  ...               1                       1         1   
4                   1  ...               0                       1         0   
...               ...  ...             ...                     ...       ...   
43190               1  ...               1                       1         0   
43191               1  ...               1                       1         0   
43192               1  ...               1                       1         0   
43193               1  ...               1                       1         0   
43194               0  ...               0                       1         1   

       gift_certificate  karaoke  free_delivery  music  pets  \
0                     0        0              0      0     0   
1                     0        0              0      0     0   
2                     0        0              0      0     0   
3                     0        0              0      0     0   
4                     1        0              0      0     0   
...                 ...      ...            ...    ...   ...   
43190                 0        1              0      0     0   
43191                 1        1              0      0     0   
43192                 1        1              1      0     0   
43193                 1        1              1      0     0   
43194                 0        0              0      0     0   

       air_conditioning  view_on_ostankino_tower  
0                     0                        0  
1                     0                        0  
2                     0                        0  
3                     0                        1  
4                     0                        0  
...                 ...                      ...  
43190                 0                        0  
43191                 0                        0  
43192                 0                        0  
43193                 0                        0  
43194                 0                        0  

[43195

<div class="alert alert-info"> <b> 2. Работа с пропущенными значениями: </b>

<b>Вызовите метод info(), чтобы увидеть число пропущенных значений</b>

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 17337 entries, 0 to 43189
Data columns (total 21 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Unnamed: 0               17337 non-null  int64  
 1   org_id                   17337 non-null  uint64 
 2   city                     17337 non-null  object 
 3   average_bill             17337 non-null  float64
 4   rating                   15573 non-null  float64
 5   rubric                   17337 non-null  object 
 6   coffee_to_go             17337 non-null  int64  
 7   food_delivery            17337 non-null  int64  
 8   breakfast                17337 non-null  int64  
 9   summer_terrace           17337 non-null  int64  
 10  wi_fi                    17337 non-null  int64  
 11  business_lunch           17337 non-null  int64  
 12  payment_by_credit_card   17337 non-null  int64  
 13  car_park                 17337 non-null  int64  
 14  gift_certificate      

<b>Очистите датасет от всех заведений, у которых средний чек неизвестен или превышает 2500</b>

In [18]:
df = df.loc[df['average_bill'] <= 2500]
df

Unnamed: 0                org_id city  average_bill    rating  \
0               1  16076540698036998306  msk         500.0  4.375000   
1               2   8129364761615040323  msk         500.0  4.000000   
2               4  13418544315327784420  msk         500.0  4.409091   
3               5   4997872545011433005  msk         500.0  4.541667   
4               8  11913692756639492916  spb         500.0  3.600000   
...           ...                   ...  ...           ...       ...   
43185       68327   9047563910052896279  spb        1000.0  3.927778   
43186       68328  17662684569129497551  spb        1000.0  4.561707   
43187       68329   5700899951016592061  msk        1500.0  4.377129   
43188       68330   4686040819909966338  msk        1500.0  3.666667   
43189       68332    971801737399231340  msk        2500.0  4.659884   

                rubric  coffee_to_go  food_delivery  breakfast  \
0      Быстрое питание             1              1          0   
1              Кофейня             1              1          1   
2              Кофейня             1              1          1   
3              Кофейня             1              1          1   
4             Столовая             0              1          0   
...                ...           ...            ...        ...   
43185         Ресторан             1              1          1   
43186         Ресторан             1              1          1   
43187       Кальян-бар             0              1          0   
43188         Ресторан             1              1          1   
43189         Ресторан             1              1          1   

       summer_terrace  ...  business_lunch  payment_by_credit_card  car_park  \
0                   0  ...               0                       1         0   
1                   1  ...               1                       0         0   
2                   1  ...               1                       0         0   
3                   1  ...               1                       1         1   
4                   1  ...               0                       1         0   
...               ...  ...             ...                     ...       ...   
43185               1  ...               1                       1         0   
43186               1  ...               1                       1         0   
43187               1  ...               1                       1         0   
43188               1  ...               1                       1         0   
43189               1  ...               1                       1         0   

       gift_certificate  karaoke  free_delivery  music  pets  \
0                     0        0              0      0     0   
1                     0        0              0      0     0   
2                     0        0              0      0     0   
3                     0        0              0      0     0   
4                     1        0              0      0     0   
...                 ...      ...            ...    ...   ...   
43185                 0        1              0      0     0   
43186                 1        1              0      1     0   
43187                 1        1              0      0     0   
43188                 1        1              0      0     0   
43189                 0        1              0      0     0   

       air_conditioning  view_on_ostankino_tower  
0                     0                        0  
1                     0                        0  
2                     0                        0  
3                     0                        1  
4                     0                        0  
...                 ...                      ...  
43185                 0                        0  
43186                 0                        0  
43187                 0                        0  
43188                 0                        1  
43189                 0                        1  

[17337

<b>Далее заполните оставшиеся пропуски средними значениями с помощью fillna()</b>

In [85]:
df = df.fillna(df.mean(numeric_only=True))
df

Unnamed: 0                org_id city  average_bill    rating  \
0               1  16076540698036998306  msk     500.00000  4.375000   
1               2   8129364761615040323  msk     500.00000  4.000000   
2               4  13418544315327784420  msk     500.00000  4.409091   
3               5   4997872545011433005  msk     500.00000  4.541667   
4               8  11913692756639492916  spb     500.00000  3.600000   
...           ...                   ...  ...           ...       ...   
43190       68333  17936653153885963296  msk    1367.98541  4.487952   
43191       68334   4379286080707082909  msk    1367.98541  3.812500   
43192       68335   7916477189329738565  msk    1367.98541  4.894231   
43193       68336  12358902585434046825  msk    1367.98541  4.156250   
43194       68338    125079240973229511  spb    1367.98541  4.633333   

                rubric  coffee_to_go  food_delivery  breakfast  \
0      Быстрое питание             1              1          0   
1              Кофейня             1              1          1   
2              Кофейня             1              1          1   
3              Кофейня             1              1          1   
4             Столовая             0              1          0   
...                ...           ...            ...        ...   
43190         Ресторан             1              1          1   
43191             Кафе             0              1          0   
43192         Ресторан             1              1          1   
43193             Кафе             1              1          1   
43194         Ресторан             0              0          0   

       summer_terrace  ...  business_lunch  payment_by_credit_card  car_park  \
0                   0  ...               0                       1         0   
1                   1  ...               1                       0         0   
2                   1  ...               1                       0         0   
3                   1  ...               1                       1         1   
4                   1  ...               0                       1         0   
...               ...  ...             ...                     ...       ...   
43190               1  ...               1                       1         0   
43191               1  ...               1                       1         0   
43192               1  ...               1                       1         0   
43193               1  ...               1                       1         0   
43194               0  ...               0                       1         1   

       gift_certificate  karaoke  free_delivery  music  pets  \
0                     0        0              0      0     0   
1                     0        0              0      0     0   
2                     0        0              0      0     0   
3                     0        0              0      0     0   
4                     1        0              0      0     0   
...                 ...      ...            ...    ...   ...   
43190                 0        1              0      0     0   
43191                 1        1              0      0     0   
43192                 1        1              1      0     0   
43193                 1        1              1      0     0   
43194                 0        0              0      0     0   

       air_conditioning  view_on_ostankino_tower  
0                     0                        0  
1                     0                        0  
2                     0                        0  
3                     0                        1  
4                     0                        0  
...                 ...                      ...  
43190                 0                        0  
43191                 0                        0  
43192                 0                        0  
43193                 0                        0  
43194                 0                        0  

[43195

<div class="alert alert-info"> <b> 3. Анализ датасета </b>

In [34]:
df = pd.read_csv('datasets_eat_places.csv')

<b>1.1 Каких заведений больше - из Москвы или Санкт-Петербурга?</b> Ответ: на 14605 больше заведений из Москвы

In [41]:
len(df[df['city'] == 'msk']) - len(df[df['city'] == 'spb'])

14605

<b>1.2 Какие типы заведений есть в датасете?</b> Ответ: 'Быстрое питание', 'Кофейня', 'Столовая', 'Бар, паб',
       'Булочная, пекарня', 'Ресторан', 'Пиццерия', 'Кафе', 'Суши-бар',
       'Кондитерская', 'Кофе с собой', 'Кальян-бар', 'Спортбар',
       'Бар безалкогольных напитков', 'Фудкорт'

In [42]:
df['rubric'].unique()

array(['Быстрое питание', 'Кофейня', 'Столовая', 'Бар, паб',
       'Булочная, пекарня', 'Ресторан', 'Пиццерия', 'Кафе', 'Суши-бар',
       'Кондитерская', 'Кофе с собой', 'Кальян-бар', 'Спортбар',
       'Бар безалкогольных напитков', 'Фудкорт'], dtype=object)

<b>2.1 Сколько в датасете суммарно ресторанов и пабов?</b> Ответ: 11345

In [55]:
bill = len(df[df['rubric'] == 'Ресторан']) + len(df[df['rubric'] == 'Бар, паб'])
bill

11345

<b>2.2 Какой у них средний чек?</b> Ответ: 2065,0581645256243

In [119]:
restorant = df[df['rubric'] == 'Ресторан']
restorant_bill = restorant['average_bill'].sum() # сумма всех чеков в ресторанах
pub = df[df['rubric'] == 'Бар, паб']
pub_bill = pub['average_bill'].sum() # cумма всех чеков в пабах
(restorant_bill + pub_bill) / bill # средний чек

2065.0581645256243

<b> 3. Какая доля заведений “Кофе с собой” действительно имеет фичу кофе на вынос? </b> Ответ: 0.3806228373702422

In [84]:
coffe_away = df[df['rubric'] =='Кофе с собой']
feature = len(coffe_away[coffe_away['coffee_to_go'] == 1])
feature / len(coffe_away)

0.3806228373702422

<b>4. Найдите разницу между средним чеком всех кафе в Москве и Санкт-Петербурге. </b> Ответ: 132.53222432978873

In [94]:
moscow = df[df['city'] == 'msk']
moscow_caffe = moscow[moscow['rubric'] == 'Кафе']
bill_moscow_caffe = moscow_caffe['average_bill'].mean()
spb = df[df['city'] == 'spb']
spb_caffe = spb[spb['rubric'] == 'Кафе']
bill_spb_caffe = spb_caffe['average_bill'].mean()
bill_moscow_caffe - bill_spb_caffe

132.53222432978873

<div class="alert alert-info"> <b> 4. Вызовите метод describe() и напишите 2-3 своих наблюдения. Например, какая фича самая редкая? </b>

In [96]:
df.describe()

Unnamed: 0        org_id  average_bill        rating  coffee_to_go  \
count  43195.000000  4.319500e+04  4.319500e+04  43195.000000  43195.000000   
mean   32609.954879  9.186257e+18  1.367985e+03      4.112360      0.414608   
std    19782.639605  5.326650e+18  3.615943e+04      0.701185      0.492660   
min        1.000000  1.148366e+13  5.000000e+02      0.000000      0.000000   
25%    16269.500000  4.570691e+18  5.000000e+02      3.952773      0.000000   
50%    31940.000000  9.159396e+18  1.367985e+03      4.125000      0.000000   
75%    49436.500000  1.377324e+19  1.367985e+03      4.523810      1.000000   
max    68338.000000  1.844596e+19  7.502000e+06      5.000000      1.000000   

       food_delivery     breakfast  summer_terrace         wi_fi  \
count   43195.000000  43195.000000    43195.000000  43195.000000   
mean        0.501678      0.312629        0.359000      0.487279   
std         0.500003      0.463570        0.479713      0.499844   
min         0.000000      0.000000        0.000000      0.000000   
25%         0.000000      0.000000        0.000000      0.000000   
50%         1.000000      0.000000        0.000000      0.000000   
75%         1.000000      1.000000        1.000000      1.000000   
max         1.000000      1.000000        1.000000      1.000000   

       business_lunch  payment_by_credit_card      car_park  gift_certificate  \
count    43195.000000            43195.000000  43195.000000      43195.000000   
mean         0.370413                0.542771      0.029679          0.056002   
std          0.482921                0.498173      0.169703          0.229928   
min          0.000000                0.000000      0.000000          0.000000   
25%          0.000000                0.000000      0.000000          0.000000   
50%          0.000000                1.000000      0.000000          0.000000   
75%          1.000000                1.000000      0.000000          0.000000   
max          1.000000                1.000000      1.000000          1.000000   

            karaoke  free_delivery         music          pets  \
count  43195.000000   43195.000000  43195.000000  43195.000000   
mean       0.166408       0.036578      0.018891      0.002454   
std        0.372451       0.187726      0.136142      0.049477   
min        0.000000       0.000000      0.000000      0.000000   
25%        0.000000       0.000000      0.000000      0.000000   
50%        0.000000       0.000000      0.000000      0.000000   
75%        0.000000       0.000000      0.000000      0.000000   
max        1.000000       1.000000      1.000000      1.000000   

       air_conditioning  view_on_ostankino_tower  
count      43195.000000             43195.000000  
mean           0.003218                 0.223174  
std            0.056636                 0.416379  
min            0.000000                 0.000000  
25%            0.000000                 0.000000  
50%            0.000000                 0.000000  
75%            0.000000                 0.000000  
max            1.000000                 1.000000

Самая редка фича - pets, среднее значение её включённости в фичи заведения - 0.002454.

Самая частая фича - payment_by_credit_card, среднее значение её включённости в фичи заведения - 0.54277

<div class="alert alert-info"> <b> 5. Создайте новую таблицу, в которой будут только заведения из Санкт-Петербурга. Ответьте на вопросы: </b>

<b>1.1 Как изменился средний чек ресторанов?</b> Ответ: уменьшился на 205.39597391817483

In [124]:
new_df = df[df['city'] == 'spb']
r_spb = new_df[new_df['rubric'] == 'Ресторан']
restorant['average_bill'].mean() - r_spb['average_bill'].mean() # переменная restorant из Анализ Датасета 2.2

205.39597391817483

<b> 1.2 Что еще можно сказать об изменениях в других столбцах? Напишите 1-2 наблюдения.</b>

In [121]:
new_df.describe()

Unnamed: 0        org_id  average_bill        rating  coffee_to_go  \
count  14295.000000  1.429500e+04  14295.000000  14295.000000  14295.000000   
mean   33234.662260  9.135664e+18   1104.686414      4.110030      0.434208   
std    19409.366514  5.336384e+18    889.253824      0.702234      0.495670   
min        8.000000  9.955441e+14    500.000000      0.666667      0.000000   
25%    16983.500000  4.515968e+18    500.000000      3.921529      0.000000   
50%    32348.000000  9.117054e+18   1367.985410      4.133333      0.000000   
75%    49807.000000  1.373255e+19   1367.985410      4.528543      1.000000   
max    68338.000000  1.844596e+19  41500.000000      5.000000      1.000000   

       food_delivery     breakfast  summer_terrace         wi_fi  \
count   14295.000000  14295.000000    14295.000000  14295.000000   
mean        0.505841      0.331724        0.382162      0.505212   
std         0.499983      0.470849        0.485933      0.499990   
min         0.000000      0.000000        0.000000      0.000000   
25%         0.000000      0.000000        0.000000      0.000000   
50%         1.000000      0.000000        0.000000      1.000000   
75%         1.000000      1.000000        1.000000      1.000000   
max         1.000000      1.000000        1.000000      1.000000   

       business_lunch  payment_by_credit_card      car_park  gift_certificate  \
count    14295.000000            14295.000000  14295.000000      14295.000000   
mean         0.395383                0.549563      0.025044          0.060511   
std          0.488950                0.497555      0.156263          0.238439   
min          0.000000                0.000000      0.000000          0.000000   
25%          0.000000                0.000000      0.000000          0.000000   
50%          0.000000                1.000000      0.000000          0.000000   
75%          1.000000                1.000000      0.000000          0.000000   
max          1.000000                1.000000      1.000000          1.000000   

            karaoke  free_delivery         music          pets  \
count  14295.000000   14295.000000  14295.000000  14295.000000   
mean       0.165233       0.028611      0.016229      0.002448   
std        0.371403       0.166718      0.126361      0.049423   
min        0.000000       0.000000      0.000000      0.000000   
25%        0.000000       0.000000      0.000000      0.000000   
50%        0.000000       0.000000      0.000000      0.000000   
75%        0.000000       0.000000      0.000000      0.000000   
max        1.000000       1.000000      1.000000      1.000000   

       air_conditioning  view_on_ostankino_tower  
count      14295.000000                  14295.0  
mean           0.002658                      0.0  
std            0.051492                      0.0  
min            0.000000                      0.0  
25%            0.000000                      0.0  
50%            0.000000                      0.0  
75%            0.000000                      0.0  
max            1.000000                      0.0

Количество заведений с фичей 'кофе на вынос' в Петербурге больше(среднее значение - 0.434208), чем количество заведений с такой же фичей сразу в двух городах: в Москве и Петербурге(среднее значение - 0.414608)

Средний рейтинг всех заведений в Петербурге меньше(4.110030), чем средний рейтинг заведений как Москвы, так и Петербурга(4.112360)